### ISSUE TITLE
Empty - non set-  application attributes are accidentaly submited by the UI

### ISSUE THREAD

**Issue by aldipower** [mesosphere/marathon#3054]
I've added a new application successfully. 

While editing the application and trying to add a command health check, I am getting this error.

Server response:
```JSON
{
  "message": "Invalid JSON",
  "details": {
    "0":{
      "path": "",
      "errors": [
        "You cannot specify both uris and fetch fields"
      ]
    }
  }
}
```

![test-invalid](https://cloud.githubusercontent.com/assets/859154/12482164/70db23ea-c04e-11e5-877b-a06e11506075.png)

App configuration:
```JSON
{
  "id":"/sleep-explotion",
  "cmd":"sleep 300",
  "args":null,
  "user":null,
  "env":{

  },
  "instances":1000,
  "cpus":0.0001,
  "mem":0.0001,
  "disk":0,
  "executor":"",
  "constraints":[

  ],
  "uris":[

  ],
  "fetch":[

  ],
  "storeUrls":[

  ],
  "ports":[
    10000
  ],
  "requirePorts":false,
  "backoffSeconds":1,
  "backoffFactor":1.15,
  "maxLaunchDelaySeconds":3600,
  "container":null,
  "healthChecks":[
    {
      "protocol":"COMMAND",
      "command":{
        "value":"exit 0"
      },
      "gracePeriodSeconds":300,
      "intervalSeconds":60,
      "timeoutSeconds":20,
      "maxConsecutiveFailures":3,
      "ignoreHttp1xx":false
    }
  ],
  "dependencies":[

  ],
  "upgradeStrategy":{
    "minimumHealthCapacity":1,
    "maximumOverCapacity":1
  },
  "labels":{

  },
  "acceptedResourceRoles":null,
  "ipAddress":null,
  "versionInfo":{
    "lastScalingAt":"2016-01-21T13:33:49.703Z",
    "lastConfigChangeAt":"2016-01-21T13:33:49.703Z"
  },
  "deployments":[

  ],
  "health":[

  ],
  "healthWeight":0,
  "lastTaskFailure":null,
  "status":2,
  "tasks":[

  ],
  "tasksRunning":0,
  "taskStats":{

  },
  "type":"CGROUP"
}
```

Marathon UI Version: 0.15-SNAPSHOT

I think the problem is that the attributes `uris`, `fetch`, `labels` are all defined in the app configuration even though they are empty. This is not correct and a serious bug.

**Comment by aldipower** [comment]
OK, I've found a difference between 0.14 and 0.15 at the service layer. So the bug wasn't introduced in the UI directly. But we have to discuss how to solve this problem.

I will provide a quickfix for now.

If I am requesting an application version, Marathon 0.15 sets an empty array for the `fetch`-attribute in the app definition. This wasn't the case in 0.14.

**Comment by pierlo-upitup** [comment]
Sorry, re-opening as I now understand the related PR is intended as a quickfix.

**Comment by pierlo-upitup** [comment]
The bugfix in 0.15.2 was not enough. When editing an app, if I disclose the Optional Settings the `uris/fetch` conflict is shown again.

**Comment by ktf** [comment]
I've this issue as well in 0.15.2. Any way this can be worked around without moving to the master?

**Comment by squirrelala** [comment]
Hi this is apparently fixed/closed for 0.15.1
However was the resolution only a UI fix? If we use the rest apis, I'm apparently still running into this issue on 0.15.1, namely trying to access
http://marathon-ip:8080/v2/apps/myapp will return json containing
app: {
...
uris: [ ],
fetch: [ ],

And then modifying this json object and attempting to post it back to update the app will get the problem above.

**Comment by toddtomkinson** [comment]
@squirrelala: I had the same experience using the salt marathon execution module and found to get around this I had to use the same fix implemented in the UI.  I remove the fetch attribute and things work again.  My PR to salt is here: saltstack/salt#32408.
